app/src/main/java/com/bagi/flowkotlin/activity/covid19/Covid19ActivityViewModel.kt

```diff
@@ -10,25 +10,29 @@ import java.text.SimpleDateFormat
 import java.util.*
 
 class Covid19ActivityViewModel : ViewModel() {
-    private val countryDetailsMLD = MutableLiveData<MutableList<Triple<Int, Int?, Int?>>>()
-    private val progressDataMLD = MutableLiveData<Boolean>()
-    private val updatedMLd = MutableLiveData<String>()
+    private val _countryDetailsMLD = MutableLiveData<MutableList<Triple<Int, Int?, Int?>>>()
+    private val _progressDataMLD = MutableLiveData<Boolean>()
+    private val _updatedMLd = MutableLiveData<String>()
 
-    fun getCountryDetailsMLD(): LiveData<MutableList<Triple<Int, Int?, Int?>>> = countryDetailsMLD
-    fun getProgressDataMLD(): LiveData<Boolean> = progressDataMLD
-    fun getUpdatedMLd(): LiveData<String> = updatedMLd
+    fun getCountryDetailsMLD(): LiveData<MutableList<Triple<Int, Int?, Int?>>> = _countryDetailsMLD
 
+    fun getProgressDataMLD(): LiveData<Boolean> = _progressDataMLD
+    fun getUpdatedMLd(): LiveData<String> = _updatedMLd
 
-    fun getCountryData(){
+    init {
+        getCountryData()
+    }
```

Comment:
I want that the triggre come from view and not from view model.

Reply:
Generally, in the MVVM architecture we want the view to be passive and to only observe changes, so it's is breaking that role. and another important thing, in your way the data will not preserve during activity kill or configurations changes(or every time the user goes to the background and comes back) and this is a major point in the ViewModel the fact that it survive and get attached to the new instance of the activity. ViewModel retain the data automatically.